apps/system/js/bluetooth_transfer.js

```diff
@@ -643,60 +643,54 @@ var BluetoothTransfer = {
+
+      var a = new MozActivity({
+        name: mappedType == 'text/vcard' ? 'import' : 'open',
+        data: {
+          type: mappedType,
+          blob: file,
+          // XXX: https://bugzilla.mozilla.org/show_bug.cgi?id=812098
+          // Pass the file name for Music APP since it can not open blob
+          filename: fileName
+        }
+      });
 
-    LazyLoader.load(['shared/js/mime_mapper.js']).then(() => {
-      getreq.onerror = () => {
-        var msg = 'failed to get file:' +
-                  filePath + getreq.error.name +
-                  getreq.error.name;
+      a.onerror = (e) => {
+        var msg = 'open activity error:' + a.error.name;
         this.debug(msg);
+        switch (a.error.name) {
+        case 'NO_PROVIDER':
+          UtilityTray.hide();
+          // Cannot identify MIMETYPE
+          // So, show cannot open file dialog with unknow media type
+          this.showUnknownMediaPrompt(fileName);
+          return;
+        case 'ActivityCanceled':
+          return;
+        case 'USER_ABORT':
+          return;
+        default:
+          return;
+        }
       };
-
-      getreq.onsuccess = () => {
-        var file = getreq.result;
-        // When we got the file by storage type of "sdcard"
-        // use the file.type to replace the empty fileType which is given by API
-        var fileName = file.name;
-        var extension = fileName.split('.').pop();
-        var originalType = file.type || contentType;
-        var mappedType = (MimeMapper.isSupportedType(originalType)) ?
-          originalType : MimeMapper.guessTypeFromExtension(extension);
-
-        var a = new MozActivity({
-          name: mappedType == 'text/vcard' ? 'import' : 'open',
-          data: {
-            type: mappedType,
-            blob: file,
-            // XXX: https://bugzilla.mozilla.org/show_bug.cgi?id=812098
-            // Pass the file name for Music APP since it can not open blob
-            filename: fileName
-          }
-        });
-
-        a.onerror = (e) => {
-          var msg = 'open activity error:' + a.error.name;
-          this.debug(msg);
-          switch (a.error.name) {
-          case 'NO_PROVIDER':
-            UtilityTray.hide();
-            // Cannot identify MIMETYPE
-            // So, show cannot open file dialog with unknow media type
-            this.showUnknownMediaPrompt(fileName);
-            return;
-          case 'ActivityCanceled':
-            return;
-          case 'USER_ABORT':
-            return;
-          default:
-            return;
-          }
-        };
-        a.onsuccess = (e) => {
-          var msg = 'open activity onsuccess';
-          this.debug(msg);
-        };
+      a.onsuccess = (e) => {
+        var msg = 'open activity onsuccess';
+        this.debug(msg);
       };
     }).catch((err) => {
+      var msg = 'failed to get file:' +
+                filePath + ' ' + err.name;
+      this.debug(msg);
```

Comment:
It is better to do the following than put these lines in the final `catch()` callback:

```javascript
Promise.all([getreq, loader])
  .then(function() {
     .... // Handling the request
  }, function(errors) {
    if (errors[0]) { // DeviceStorage Error
      var msg = 'failed to get file:' + filePath + ' ' + errors[0].name;
      this.debug(msg);
    }
    if (errors[1]) { // LazyLoader Error
      throw errors[1];
    }
  })
  .catch(function(e) {
    console.error(e);
  });
```

By using the second `onReject` callback of `then()` method, you would retain the opportunity to handle the deviceStorage error separately.

The final `catch()` will also help us print out TypeError if there is any typo in any (rejected or resolved) functions in the chain.

It is also important to recognize a promise chain is also an error handling chain. You should always try to handle the error in your own step in the chain and throw the error to the next `catch` callback if you couldn't handle them -- or, handle the error but still throw to abort all the steps follows.